### ISSUE TITLE
[leo_statistics] leo_(gateway|storage|manager) can start normally even if starting statistics related erlang processes failed

### ISSUE THREAD

**Issue by mocchira** [leo-project/leofs#361]
For example,
https://github.com/leo-project/leo_storage/blob/develop/src/leo_storage_app.erl#L92

The above caller expect an Exception thrown from a callee,
when invokindg *:start_link failed, 
but they are not throwing an Exception but returning an error tuple.

As a result, leo_(gateway|storage|manager) can start normally even if starting statistics related erlang processes failed.

**Comment by yosukehara** [comment]
I've found some dead leo_statistics related processes, which is rare case when using low performance disk (on VM).

### normal case:
```erlang
$ ./package/leo_manager_0/bin/leo_manager remote_console

(manager_0@127.0.0.1)1> supervisor:which_children(savanna_commons_sup).
                            [{'sv_vm_stats_5m.vm-used-per-allocated-mem',<0.1377.0>,worker,[svc_metric_server]},


(manager_0@127.0.0.1)2> supervisor:which_children(savanna_commons_sup).
[{'sv_vm_stats_5m.vm-used-per-allocated-mem',<0.1377.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_5m.vm-total-mem',             <0.1376.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_5m.vm-system-mem',            <0.1375.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_5m.vm-procs-mem',             <0.1374.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_5m.vm-proc-count',            <0.1373.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_5m.vm-ets-mem',               <0.1372.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_5m.vm-allocated-mem',         <0.1371.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_1m.vm-used-per-allocated-mem',<0.1368.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_1m.vm-total-mem',             <0.1367.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_1m.vm-system-mem',            <0.1366.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_1m.vm-procs-mem',             <0.1365.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_1m.vm-proc-count',            <0.1364.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_1m.vm-ets-mem',               <0.1363.0>,worker,[svc_metric_server]},
 {'sv_vm_stats_1m.vm-allocated-mem',         <0.1362.0>,worker,[svc_metric_server]},
 {folsom,<0.1325.0>,supervisor,[folsom]}]
```

### failure case
```erlang
$ ./package/leo_manager_0/bin/leo_manager remote_console

(manager_0@127.0.0.1)1> supervisor:which_children(savanna_commons_sup).
                            [{'sv_vm_stats_5m.vm-used-per-allocated-mem',<0.1377.0>,worker,[svc_metric_server]},


(manager_0@127.0.0.1)2> supervisor:which_children(savanna_commons_sup).
[{folsom,<0.1325.0>,supervisor,[folsom]}]
```

**Comment by yosukehara** [comment]
* related a ticket: https://github.com/leo-project/leofs/issues/274

**Comment by PatelParas** [comment]
great  :+1:

**Comment by yosukehara** [comment]
I'll check leo_gateway and leo_storage soon.

**Comment by yosukehara** [comment]
I've fixed this issue, which will be included in LeoFS v1.2.9.

* leo_gateway: https://github.com/leo-project/leo_gateway/commit/d862763e385ba110a83cfff01a53a21b51d9ed87
* leo_storage: https://github.com/leo-project/leo_storage/commit/68000cfed976b8cb0570397653db1bcbb03688cc
